addons/website/static/src/components/dialog/seo.js

```diff
@@ -369,13 +591,169 @@ class TitleDescription extends Component {
     }
 }
 
+export class SeoChecks extends Component {
+    static template = "website.SeoChecks";
+    static components = {
+        CheckBox,
+    };
+    static props = {};
+
+    setup() {
+        this.website = useService("website");
+        this.seoContext = useState(seoContext);
+        this.state = useState({
+            altAttributes: this.getAltAttributes(),
+            checkingLinks: false,
+            checkingLink: false,
+            checkedLinks: false,
+            counterLinks: 0,
+            totalLinks: 0,
+            headingsScan: this.getHeadingsScan(),
+        });
+        this.imgUpdated = this.imgUpdated.bind(this);
+    }
+
+    imgUpdated(img) {
+        img.updated = true;
+        this.seoContext.updatedAlts = this.state.altAttributes.filter(img => img.updated);
+    }
+
+    getHeadingsScan() {
+        const ret = {
+            missingH1: false,
+            multipleH1: false,
+            misplacedH1: false,
+        };
+        const allHeadingsEls = Array.from(
+            this.website.pageDocument.documentElement.querySelectorAll("#wrap h1,h2,h3,h4,h5,h6")
+        );
+        const h1Els = allHeadingsEls.filter((heading) => heading.tagName.toLowerCase() === "h1");
+
+        if (h1Els.length === 0) {
+            // We must have a h1 tag
+            ret.missingH1 = true;
```

Comment:
Question: should we display the "H1 tag is missing..." warning for empty pages ?

Reply:
If a page is empty and published, it's bad for SEO. So IMO we should warn the customer.

Reply:
I agree with Serge